File: cpp/bc1045.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main()
{
	float cVal[3], aux;
	
	cin >> cVal[0] >> cVal[1] >> cVal[2];
	
	while(true)
	{
		if(cVal[1] < cVal[2])
		{
			aux = cVal[1];
			cVal[1] = cVal[2];
			cVal[2] = aux;
		}
		else if(cVal[1] > cVal[0])
		{
			aux = cVal[1];
			cVal[1] = cVal[0];
			cVal[0] = aux;
		}
		else break;
		
	}
	
	if(cVal[0] >= cVal[1] + cVal[2]) cout << "NAO FORMA TRIANGULO" << endl;
	else
	{
		float fVer = pow(cVal[0], 2);
		float sVer = pow(cVal[1], 2) + pow(cVal[2], 2);
		
		if(fVer == sVer) cout << "TRIANGULO RETANGULO" << endl;
		else if(fVer > sVer) cout << "TRIANGULO OBTUSANGULO" << endl;
		else if(fVer < sVer) cout << "TRIANGULO ACUTANGULO" << endl;
		
		if(cVal[0] == cVal[1] && cVal[1] == cVal[2]) cout << "TRIANGULO EQUILATERO" << endl;
		else if((cVal[0] == cVal[1] && cVal[1] != cVal[2]) || (cVal[0] != cVal[1] && cVal[1] == cVal[2])) cout << "TRIANGULO ISOSCELES" << endl;
	}
}

File: python/bc1045.py
from math import pow

cVal = input().split(" ")

valA = float(cVal[0])
valB = float(cVal[1])
valC = float(cVal[2])

while True:
    if valB > valA:
        aux = valB
        valB = valA
        valA = aux
    elif valB < valC:
        aux = valB
        valB = valC
        valC = aux
    else: break

if valA >= valB + valC: print("NAO FORMA TRIANGULO")
else:
    fRes = pow(valA, 2)
    sRes = pow(valB, 2) + pow(valC, 2)

    if fRes == sRes: print("TRIANGULO RETANGULO")
    elif fRes > sRes: print("TRIANGULO OBTUSANGULO")
    elif fRes < sRes: print("TRIANGULO ACUTANGULO")

    fVer = valA == valB and valB == valC
    sVer = (valA == valB and valB != valC) or (valA != valB and valB == valC)

    if fVer == True: print("TRIANGULO EQUILATERO")
    elif sVer == True: print("TRIANGULO ISOSCELES")
